### PR TITLE
[1.3.6-prepare][Fix-#4840][worker] fix master fault tolerance when startup #4845

### PR DESCRIPTION
**Pull request by CalvinKirs**
issue #4840 
pr #4845

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java [hunk 1]
```diff
@@ -72,6 +79,9 @@ public void start() {
 			mutex = new InterProcessMutex(getZkClient(), znodeLock);
 			mutex.acquire();
 
+            //  Master registry
+            masterRegistry.registry();
+
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java:86]: Indent problem

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java [hunk 2]
```diff
@@ -94,6 +104,7 @@ public void start() {
 	@Override
 	public void close(){
 		super.close();
+        masterRegistry.unRegistry();
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java:109]: Indent problem

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java [hunk 3]
```diff
@@ -160,6 +171,9 @@ private void removeZKNodePath(String path, ZKNodeType zkNodeType, boolean failov
 	 * @throws Exception	exception
 	 */
 	private void failoverServerWhenDown(String serverHost, ZKNodeType zkNodeType) throws Exception {
+        if (StringUtils.isEmpty(serverHost)) {
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java:176]: Indent problem

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java [hunk 4]
```diff
@@ -272,9 +286,8 @@ private boolean checkTaskAfterWorkerStart(TaskInstance taskInstance) {
 
 		if(workerServerStartDate != null){
 			return taskInstance.getStartTime().after(workerServerStartDate);
-		}else{
-			return false;
-		}
+        }
+		return false;
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java]: Indent problem

**Review by chengshiwen — CHANGES_REQUESTED** [review]

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java (hunk 4 above)
Comment (CalvinKirs) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/zk/ZKMasterClient.java]: hi, thanks for your review , I have resolved the code style, please take a look.

**Review by lenboo — APPROVED** [review]
+1

**Pull request merged**